### PR TITLE
Add DEBUG_RCVR_LINKSTATS

### PR DESCRIPTION
**Pull request by CapnBry**
Adds a new `DEBUG_RCVR_LINKSTATS` define which dumps RX linkstats on every packet to Serial. This is intended to allow anyone to create the comparisons I've been doing with recording every single packet's RF information for analysis. Note that for testing multiple simultaneous receivers with this, the TLM Ratio should be set to Off to prevent everyone from transmitting over each other. `DEBUG_CRSF_NO_OUTPUT` should probably be enabled too, else the CRSF will mix in with the log.

### Transmitter
When enabled on the transmitter, this converts channels CH1-CH4 into a 32 bit packet ID counter that increments once for every channels packet sent. This allows each RX to compare exactly the same packet across all receivers.

### Receiver
On the receiver side, this debug adds a dump of the RF stats to `Serial`, regardless of if `DEBUG_LOG` is set or not. The regular debug log messes up processing of the CSV file this can generate. In addition, regular INFOLN and ERRLN calls are also disabled to prevent them from being added to the log as well. This is purely for getting info on every packet received.

Enabling this debug mode disables LPF smoothing on the RSSI values, so the true per-packet RSSI is reported. 

The columns in the CSV reported are:
| Field | Value | Decscription |
|----|----|----|
|0| PacketID | The sequential packet counter from the transmitter, this should increase by one for every row output, since the counter only increments when sending this type of packet |
|1| Antenna | 0 or 1 for active antenna |
|2| RSSI | The LastPacketRSSI for the packet, on whichever antenna received it |
|3| LQ| Link Quality at packet reception |
|4| SNR | LastPacketSNR |
|5| TX Power | One of our `PowerLevels_e` power indexes from the transmitter, requires Wide switch mode, may lag up to 8 packets from actual power change|
|6| FHSS | The FHSS hop channel index the packet was received on |
|7|TimingOffset| Receive time of the packet relative to our expected RX time (microseconds)

## Incidental Changes
I refactored `getRFlinkInfo` in rx_main a little so that the LPF could be disabled. The functionality should be identical except it is 56 bytes less code-- SOFAST! The one difference is that the `downlink_*` fields of the LinkStats are not zeroed every packet reception now. They are already zero at start, and never get changes so why waste CPU time setting them 500x a second.

### CODE REVIEW

File: src/user_defines.txt [hunk 1]
```diff
@@ -104,3 +104,9 @@
 
 # These debugging options send extra information to BetaFlight in the LinkStatistics packet
 #-DDEBUG_BF_LINK_STATS
+
+# Prints a log line for every channels packet recieved at the RX: ID,Antenna,RSSI,LQ,SNR,PWR,FHSS
```
Comment (pkendall64) [inline src/user_defines.txt:108]: and `PFD`

File: src/src/rx_main.cpp [hunk 2]
```diff
@@ -1163,6 +1165,34 @@ static void checkSendLinkStatsToFc(uint32_t now)
...
+        crsfLinkStatistics_t ls = *(crsfLinkStatistics_t *)((const void *)&crsf.LinkStatistics);
+        uint32_t packetCounter = debugRcvrLinkstatsPacketId;
+        uint8_t fhss = debugRcvrLinkstatsFhssIdx;
+        // actually the previous packet's offset since the update happens in tick, and this will
+        // fire right after packet reception (a little before tock)
+        int32_t pfd = PfdPrevRawOffset;
+
+        // Use serial instead of DBG() because do not necessarily want all the debug in our logs
+        char buf[50];
+        snprintf(buf, sizeof(buf), "%u,%u,-%u,%u,%d,%u,%u,%d\r\n",
```
Comment (pkendall64) [inline src/src/rx_main.cpp:1186]: I think we need to be able to enable debug logging like this where just the debug lines for a particular "feature" are printed. But I'm ok with this currently.

File: src/user_defines.txt (hunk 1 above)
Comment (CapnBry) [reply src/user_defines.txt]: Thank you very much!

File: src/src/rx_main.cpp (hunk 2 above)
Comment (CapnBry) [reply src/src/rx_main.cpp:1186]: Yeah like some sort of module-grouped logging defines would be nice if we could work out some major functionality groups to put things in.

**Review by pkendall64 — APPROVED** [review]

**Review by JyeSmith — APPROVED** [review]

**Pull request merged**